### PR TITLE
Support PCM -> fbank transform inside tensorflow

### PR DESCRIPTION
**Pull request by paulo-raca**
I'm attempting to export the model and use it in a non-python app.

While the tensorlow model is quite portable, the input pre-processing (audio.mfcc_fbank) is not.
    
This PR adds support to moving this stage into tensorflow by adding `pcm_input=True` when building the model.
    
When this is set, the input becomes a tensor with raw PCM data and shape [batch_size, num_samples].
    
The implementation attempts to be as close as possible to the original, however due to the different implementations/libraries, the results do differ a little bit.
    
The PR also provides `example_pcm.py`, a modified version of example.py which uses raw PCM inputs and batched tensorflow operations

### CODE REVIEW

**Comment by philipperemy** [comment]
@paulo-raca thank you it looks good!

https://github.com/philipperemy/deep-speaker/pull/84/files#diff-2a959e987f602089b7e173a24f237d0c75db4848ad25875d9880e43706235898R61

In this one, NUM_FRAMES was not None before and now it's None. Can it be a problem?

**Comment by paulo-raca** [comment]
> In this one, NUM_FRAMES was not None before and now it's None. Can it be a problem?

The model works well even if we use it on audio files with different lengths

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ jobs:
         # stop the build if there are Python syntax errors or undefined names
         flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
         # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
-        flake8 . --count --max-complexity 10 --max-line-length 127 --statistics
+        flake8 . --count --ignore=E203 --max-complexity 10 --max-line-length 127 --statistics
```
Comment (paulo-raca) [inline .github/workflows/ci.yml:34]: flake8 and black checks disagree about spaces before collons. This disables the flake8 check and uses black style

**Comment by philipperemy** [comment]
@paulo-raca Thanks for the headsup! I'll have a look and merge it ;)

**Comment by paulo-raca** [comment]
Thanks!

**Pull request merged**